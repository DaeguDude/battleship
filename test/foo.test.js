test("", () => {
  const myMock = jest.fn().mockName("a mock name");

  myMock(1, 2);

  expect(myMock).toHaveBeenCalled();
  expect(myMock.mock.calls.length).toBeGreaterThan(0);

  expect(myMock).toHaveBeenCalledWith(1, 2);
  expect(myMock.mock.calls).toContainEqual([1, 2]);

  expect(myMock).toHaveBeenLastCalledWith(1, 2);
  expect(myMock.mock.calls[myMock.mock.calls.length - 1]).toEqual([1, 2]);

  expect(myMock.mock.calls[myMock.mock.calls.length - 1][0]).toBe(1);

  expect(myMock.mock.calls).toEqual([[1, 2]]);
  expect(myMock.getMockName()).toBe("a mock name");
});
