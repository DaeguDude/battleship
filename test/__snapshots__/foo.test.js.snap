// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
    Array [
      1,
      2,
    ],
    Array [
      1,
      2,
    ],
    Array [
      1,
      2,
    ],
    Array [
      "yay",
      "ho",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
